dev-java/cglib/cglib-3.3.0.ebuild

```diff
@@ -0,0 +1,50 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+JAVA_PKG_IUSE="source test"
```

Comment:
should have also `doc` flag

Reply:
With USE=doc it would run into errors -> [build.log.gz](https://github.com/gentoo/gentoo/files/6210510/build.log.gz)
How to proceed?  Are there docs at all?  If so I could not find them :-(  
Also an "examples" USE flag seems not to trigger any examples' installation.

Regarding the slot dependencies I successfully raised asm to `:9`.  Is possible to drop the slot restriction or to set a `minimal slot` so it could use any slot >7?  I've never seen such case :-(

> this one might not be needed, might be a bug in `java-ebuilder`

Removal of the `"${MY_P}"`line seems not to break anything.

> @vaukai might also want to change the pr subject

PR subject of course will be synced with latest commit message.

Before I push again, could we please solve the USE flags matter.  Meanwhile I need to try and understand your comments on the tests stuff ....

Reply:
wrt min slot, it's safer to use `:9` as future version of `asm` does not have to be compatible with `cglib`.

Reply:
wrt the `doc` use flag, can you try to run following command in the `WORKDIR` after emerge fails?
```
javadoc -Xdoclint:none -d target/api -encoding UTF-8 -docencoding UTF-8 -charset UTF-8 -classpath //usr/share/asm-9/lib/asm.jar://usr/share/ant-core/lib/ant.jar://usr/share/ant-core/lib/ant-bootstrap.jar://usr/share/ant-core/lib/ant-launcher.jar -quiet @sources.lst
```
then you can try to add the generated `cglib.jar` file to the `-classpath` and see whether it helps. because the error says `package net.sf.cglib does not exist` so i suppose it might need to have that on the classpath, though not sure, i'm not that familiar with `javadoc`.

also, the api documentation in java is inlined in the source files so it's generated from the source files. jfyi :-)

Reply:
This is what happens:

```
/var/tmp/portage/dev-java/cglib-3.3.0/work $ javadoc -Xdoclint:none -d target/api -encoding UTF-8 -docencoding UTF-8 -charset UTF-8 -classpath //usr/share/asm-9/lib/asm.jar://usr/share/ant-core/lib/ant.jar://usr/share/ant-core/lib/ant-bootstrap.jar://usr/share/ant-core/lib/ant-launcher.jar -quiet @sources.lst
cglib-RELEASE_3_3_0/cglib/src/main/java/net/sf/cglib/reflect/MethodDelegate.java:20: error: package net.sf.cglib does not exist
import net.sf.cglib.*;
^
1 error
```
Nothing gets generated there.  How is the procedure to compile cglib outside of portage including the javadocs?